### ISSUE TITLE
Text Extraction / How to get font information like italic or bold

### ISSUE THREAD

**Issue by TuningGuide** [mozilla/pdf.js#7372]
Hi, I'm sorry to misuse this issue reporting for a question. But I guess I get the most usefull information from here.

If I use

    let content = yield page.getTextContent();
    content.items[0]...

How can I tell that item 0 is italic or bold?

**Comment by timvandermeij** [comment]
As far as I know this information is not exposed through the API. Marking as a feature for now.

**Comment by Snuffleupagus** [comment]
In general, I don't think that it's actually possible to obtain this information with any certainty, considering that there's a number of ways a font could be made to look bold/italic.

For example, in non-embedded *standard* fonts, you can often tell by the font name ([as is done here](https://github.com/mozilla/pdf.js/blob/7c7d2399956c25491aa7d2548c9c784cc2b3cea8/src/core/fonts.js#L535-L537)).
For embedded fonts, it's usually a lot more difficult to tell, since the font name may be (and often is) somewhat arbitrary. In this case, whether or not the text looks bold/italic depends on the embedded font data itself.
Then there are the [font flags](https://github.com/mozilla/pdf.js/blob/7c7d2399956c25491aa7d2548c9c784cc2b3cea8/src/core/fonts.js#L101-L111), but based on experience I wouldn't really trust them, since PDF files unfortunately often provides incomplete/inconsistent/broken font information.
Furthermore, even though (in my experience) a majority of PDF files uses different fonts to represent regular/bold/italic text, there are exceptions. One such example is #7297, where the `Text rendering mode` is modified in order to simulate bold text.

So, to summarize, there's unfortunately no way that you can consistently obtain the information you're asking for. And as https://github.com/mozilla/pdf.js/issues/7297#issuecomment-217631260 said, this seems out-of-scope for text-selection/text-extraction use case.

**Comment by TuningGuide** [comment]
@Snuffleupagus thank you for the insights. I'l have a look at the font names used, and hardcode it
